AgilityShortcutTest

wvanderp opened the pull request:
for the new message see below

---

I was inspired by https://github.com/Skretzo/shortest-path/issues/54#issue-2019897329 , which asked why we did not use the built-in list of agility shortcuts.

The question was answered there, but the thought struck me that we could use a unit test to check if we have all the agility shortcuts.

So I made a test. It resulted in about 100 missing (of a total of around 280).

This unit test will fail right now, so the best course of action is to slowly start adding these shortcuts either to the shortcuts.tsv or to the ignore list in the test if there is a valid reason for it.

I hope to be slowly chipping away at the list, although some of it will be hard because I can't visit the area or take the shortcuts.

---

https://github.com/Skretzo/shortest-path/pull/255

I had been slowly working on adding all the agility shortcuts, but you know how life sometimes gets in the way.

I still did not manage to add them all; many need a high agility to test them.
But I do think that it's time to merge the things I have now.

There are a few extra things that are in there, like a unit test that looks at the runelite enum for the number of agility shortcuts.
A unit test for graple hooks, I thought it was not working, but it was, and now there is a unit test.
Also, an extra tsv linting rule for finding extra spaces around values.
and lastly, many agility shortcuts.

As always, if anything is not to our liking, I am willing to change it.

this PR is ready to be reviewed and merged

Skretzo commented:
This is a good idea!

It is perfectly ok to add shortcuts you have not tested in-game. I try to check videos if possible or estimate the time it takes to use the shortcut based on the distance in number of tiles between start and end. [Mejmap](https://mejrs.github.io/osrs?m=-1&z=5&p=0&x=3139&y=3515&layer=nomove&layer=objects&layer=crowdsourcetransports) is very useful for this, as it allows you to see objects and even crowdsourced transports and teleports (although not up-to-date anymore). When you hover or click the transport or tele pin you see the start and end tile and information about the transport (e.g. which object menu option was clicked, the object id, and sometimes which varbit controls the state of the object). Beware that it can be wrong in some cases, e.g. the start or end tile does not make sense and was somehow collected incorrectly.

wvanderp committed:
Refactor transport data files and add unit tests for agility shortcuts

- Updated seasonal_transports.tsv and transports.tsv to improve formatting and add new entries.
- Introduced AgilityShortcutTest to verify that all AgilityShortcut enum constants have corresponding TSV entries.
- Added ObservatoryGrappleShortcutTest to validate the transport requirements and properties for the Observatory Grapple Shortcut.
- Enhanced transport data extraction methods to ensure comprehensive coverage of agility shortcuts and their requirements.

FIrgolitsch commented:
Happy to see you found the time to get this done! Looks good to me, merging it now.
